### ISSUE TITLE
Consider using VS Code watches for all URIs that VS Code supports

### ISSUE THREAD

**Issue by DavyLandman** [usethesource/rascal-language-servers#266]
In LSP 3.17 there is a whole part of file watches added: https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#workspace_didChangeWatchedFiles

We might be able to use that instead of our own watches. This concerns both our added VFS implementation as our internal File Watches for example, which we know on OSX can lag behind.

**Comment by jurgenvinju** [comment]
Interesting thought but I've no idea how to accomplish that transparently without also cloning the rest of the scheme implementations and without breaking the code that runs without VScode.

How about we integrate these triggers into the current triggers? That way we know sooner when things have changed.

**Comment by DavyLandman** [comment]
Yeah, true, this would be truly inside out to defer something like file scheme to VS Code.. Maybe this is too far fetched. Or maybe we can do special where we can have watch providers registering itself?

**Comment by DavyLandman** [comment]
#967 partially does this, but only of files not already supported by rascal itself, stuff gets too messy otherwise.
